README.md

```diff
@@ -1,486 +1,679 @@
-
-### `@bound`
-
-The `@bound` decorator makes a method auto-bound: it will carry around the original `this` value when accessed as `this.method` and not immediately called. This behavior matches Python's semantics, and it's been found useful in the React ecosystem, which makes frequent use of passing functions around. Example usage:
+Decorators can make this possible by giving someone access to a private field or method. This may be encapsulated in a "private key"--an object which contains these references, to be shared only with who's appropriate.
 
 ```mjs
-import { @bound } from "./bound.mjs";
+import { PrivateKey } from "./private-key.mjs"
 
-class Foo {
-  x = 1;
+let key = new PrivateKey;
 
-  @bound method() { console.log(this.x); }
+export class Box {
+  @key.show #contents;
+}
 
-  queueMethod() { setTimeout(this.method, 1000); }
+export function setBox(box, contents) {
+  return key.set(box, contents);
 }
 
-new Foo().queueMethod();  // will log 1, rather than undefined
+export function getBox(box) {
+  return key.get(box);
+}
 ```
 
-One possible implementation, based on `@register`:
+Note that this is a bit of a hack, and could be done better with constructs like references to private names with [`private.name`](https://gist.github.com/littledan/ab73ff08f98f33088a0072ad202445b1) and broader scope of private names with [`private`/`with`](https://gist.github.com/littledan/5451d6426a8ed65c0f3c2822c51314d1). But it shows that this decorator proposal "naturally" exposes existing things in a useful way.
 
 ```mjs
-// bound.mjs
-export decorator @bound {
-  @register((target, name) => {
-    const method = target[name];
-    Object.defineProperty(target, name, {
-      get() {
-        const bound = method.bind(this);
-        Object.defineProperty(this, name, { value: bound, configurable: true });
-      }
-      configurable: true
-    });
-  })
+// private-key.mjs
+export class PrivateKey {
+  #get;
+  #set;
+  
+  show({get, set}) {
+    assert(this.#get === undefined && this.#set === undefined);
+    this.#get = get;
+    this.#set = set;
+    return {get, set};
+  }
+  get(obj) {
+    return this.#get(obj);
+  }
+  set(obj, value) {
+    return this.#set(obj, value);
+  }
 }
 ```
 
-There are various approaches to writing an auto-bound decorator, but ultimately, the most efficient way may be built into the JavaScript engine; see [NEXTBUILTINS.md](./NEXTBUILTINS.md#bound) for discussion of a built-in `@bound` decorator, and the [bound-decorator](https://github.com/mbrowne/bound-decorator) repository for another approach.
+This example could be roughly desugared as follows:
 
-### `@callable`
+```mjs
+let initialize, get, set;
+export class Box {
+  #_contents = initialize(undefined);
+  get #contents() { return get.call(this); }
+  set #contents(v) { set.call(this, v); }
+
+  static #_ = (
+    get = function() { return this.#_contents; },
+    set = function(v) { this.#_contents = v; }
+  )
```

Comment:
this seems strange, is it a hack around the lack of static blocks? if so, perhaps a comment and link to that proposal would help add clarity.

Reply:
Yes, it's a hack around the lack of static blocks, exactly. I thought the audience wouldn't be familiar with them. I'd be happy to accept a quick fix to use static blocks when appropriate, if anyone wants to suggest the change.

Reply:
that's a reasonable expectation, but this pattern is quite confusing as it is, it took me a bit to figure out what it was doing. I think a comment would be helpful, even for now.

Reply:
PRs welcome

Reply:
OK, I've replaced this with a static block, if that will be more clear to people.

Reply:
Oh I was confused with the static block; thought I missed some new syntax. For me, the original variant is pretty clear even if it is a hack. I would agree with @ljharb: the comment should be enough.